stl/inc/syncstream

```diff
@@ -0,0 +1,355 @@
+            if (!emit()) {
+                return -1;
+            }
+        }
+        return 0;
+    }
+
+    int_type overflow(int_type _Current_elem) override {
+        if (_Wrapped == nullptr) {
+            return _Traits::eof();
+        }
+        const bool _Chk_eof = _Traits::eq_int_type(_Current_elem, _Traits::eof());
+        if (_Chk_eof) {
+            return _Traits::not_eof(_Current_elem);
+        }
+
+        const size_type _Buf_size       = _Get_buffer_size();
+        const size_type _Max_allocation = allocator_traits<_Alloc>::max_size(_Al);
+        if (_Buf_size == _Max_allocation) {
+            return _Traits::eof();
+        }
+
+        const size_type _New_capacity  = _Calculate_growth(_Buf_size, _Buf_size + 1, _Max_allocation);
+        const _Elem* const _Old_ptr    = streambuf_type::pbase();
+        const size_type _Old_data_size = _Get_data_size();
+
+        _Elem* const _New_ptr = _Unfancy(_Al.allocate(_New_capacity));
+        _Traits::copy(_New_ptr, _Old_ptr, _Old_data_size);
+
+        streambuf_type::setp(_New_ptr, _New_ptr + _Old_data_size, _New_ptr + _New_capacity);
+        streambuf_type::sputc(_Traits::to_char_type(_Current_elem));
+
+        return _Current_elem;
+    }
+
+private:
+    static constexpr size_type _Min_size = 32; // constant for minimum buffer size
+
+    void _Init() {
+        _Elem* const _New_ptr = _Unfancy(_Al.allocate(_Min_size));
+        streambuf_type::setp(_New_ptr, _New_ptr + _Min_size);
+    }
+
+    void _Tidy() noexcept {
+        const size_type _Buf_size = _Get_buffer_size();
+        if (0 < _Buf_size) {
+            _Al.deallocate(pointer_traits<pointer>::pointer_to(*streambuf_type::pbase()), _Buf_size);
+        }
+
+        streambuf_type::setp(nullptr, nullptr, nullptr);
+        if (_Wrapped != nullptr) {
+            _Release_mutex_for_instance(_Wrapped);
+            _Wrapped = nullptr;
+        }
+    }
+
+    void _Move_assign(basic_syncbuf&& _Right, _Equal_allocators) noexcept {
+        _Tidy();
+        _Pocma(_Al, _Right._Al);
+        _Swap_except_al(_Right);
+    }
+
+    void _Move_assign(basic_syncbuf&& _Right, _Propagate_allocators) noexcept {
+        _Tidy();
+        _Pocma(_Al, _Right._Al);
+        _Swap_except_al(_Right);
+    }
+
+    void _Move_assign(basic_syncbuf&& _Right, _No_propagate_allocators) {
+        if (_Al == _Right._Al) {
+            _Move_assign(_STD move(_Right), _Equal_allocators{});
+        } else {
+            _Tidy();
+
+            const size_type _Right_buf_size  = _Right._Get_buffer_size();
+            const size_type _Right_data_size = _Right._Get_data_size();
+
+            _Elem* const _New_ptr = _Unfancy(_Al.allocate(_Right_buf_size));
+            _Traits::copy(_New_ptr, _Right.pbase(), _Right_data_size);
+
+            streambuf_type::setp(_New_ptr, _New_ptr + _Right_data_size, _New_ptr + _Right_buf_size);
+            streambuf_type::_Plocale = _STD move(_Right._Plocale);

```

Comment:
You are moving the pointer rather than swapping it, so it's deallocated twice. This causes `test_syncbuf_move_swap_operations<non_move_assignable_non_equal_allocator<char>>` to fail (took _only_ an hour to find).
```suggestion
            _STD swap(streambuf_type::_Plocale, _Right._Plocale);
```

Reply:
Good catch!
That's a subtle one to find, I would have never guessed 😢.